### PR TITLE
MRTI-3616 # split big input files and quiet output by default

### PR DESCRIPTION
**Pull request by samah17**

### CODE REVIEW

File: datalake_scripts/scripts/bulk_lookup_threats.py [hunk 1]
```diff
@@ -131,13 +143,12 @@ def main(override_args=None):
         hashkey_only=args.hashkey_only
     )
 
-    pretty_print(response, args.output_type)
-
     if args.output:
-        starter.save_output(args.output, response)
+        starter.save_output(args.output, response,request_index)
```
Comment (juan-kabbali) [inline datalake_scripts/scripts/bulk_lookup_threats.py:147]: missing white space after 'comma'  `starter.save_output(args.output, response,request_index)`

File: datalake_scripts/scripts/bulk_lookup_threats.py [hunk 2]
```diff
@@ -94,12 +94,24 @@ def main(override_args=None):
         for input_file in args.input:
             file_atom_type, filename = get_atom_type_from_filename(input_file)
             logger.debug(f'file {filename} was recognized as {file_atom_type}')
-
+            size_limit = 500
```
Comment (juan-kabbali) [inline datalake_scripts/scripts/bulk_lookup_threats.py:97]: This could be defined outside the `for` to avoid re-declaring it at each loop

**Review by juan-kabbali — APPROVED** [review]

File: datalake_scripts/common/base_script.py [hunk 3]
```diff
@@ -106,6 +108,12 @@ def _load_list(self, file_name: str) -> list:
         """
         return [line.rstrip('\n') for line in open(file_name)]
 
+    def _split_input_file(self, input_file, n):
```
Comment (Hugo-C) [inline datalake_scripts/common/base_script.py:111]: no need for the `_` prefix if it is used by external code

**Review by Hugo-C — CHANGES_REQUESTED** [review]

File: datalake_scripts/scripts/bulk_lookup_threats.py [hunk 4]
```diff
@@ -87,29 +90,49 @@ def main(override_args=None):
 
     # validate that at least there is one untyped atom or one atom or one input file
     if (not has_flag and not has_file and not args.untyped_atoms) or (SUBCOMMAND_NAME in args.untyped_atoms):
-        parser.error("you must provide at least one of following: untyped atom, atom type, input file.")
+        parser.error(
+            "you must provide at least one of following: untyped atom, atom type, input file.")
 
     # process input files
     if has_file:
+        size_limit = 500
```
Comment (Hugo-C) [inline datalake_scripts/scripts/bulk_lookup_threats.py:98]: With the command:
```
bulk_lookup_threats.py -i ip:input.text reverso.net --domain paiza.com --ip 45.96.65.132 -o output.json -ad -e preprod
```
(I took the IP from https://raw.githubusercontent.com/nl8590687/spam-list/master/spam-ip-list.txt)
I got :
```
API returned non 2xx response code : 429
{"message":"Number of threats to lookup is over request limit (max 500)"}
```
I don't know what the issue is :/

File: datalake_scripts/common/base_script.py [hunk 5]
```diff
@@ -26,7 +26,7 @@ def save_output(self, file_name: str, data):
         If data is a list, file format will be txt.
         Else it will be saved as it comes.
         """
-        with open(file_name, 'w+') as file_to_write:
+        with open(file_name, 'a+') as file_to_write:
```
Comment (Hugo-C) [inline datalake_scripts/common/base_script.py:29]: Please rollback if no longer needed

**Review by Hugo-C — APPROVED** [review]

**Pull request merged**